src/indexed_crate.rs

```diff
@@ -148,3 +156,73 @@ fn collect_public_items<'a>(
         Visibility::Crate | Visibility::Restricted { .. } => {}
     }
 }
+
+fn create_dummy_trait_items(crate_: &Crate) -> HashMap<Id, Item> {
+    let paths = crate_
+        .index
+        .values()
+        .map(|item| &item.inner)
+        .filter_map(|item_enum| match item_enum {
+            rustdoc_types::ItemEnum::Impl(impl_) => Some(impl_),
+            _ => None,
+        })
+        .filter_map(|impl_| impl_.trait_.as_ref());
+
+    // Limiting the creation of dummy traits to only those that are used by the lints.
+    // There are other foreign traits and it is not obvious how the dummy traits
+    // should look like for them.
+    let derivable_traits = [
+        "Debug",
+        "Clone",
+        "Copy",
+        "PartialOrd",
+        "Ord",
+        "PartialEq",
+        "Eq",
+        "Hash",
+        "Default",
+    ];
+    let auto_traits = ["Send", "Sync", "Unpin", "RefUnwindSafe", "UnwindSafe"];
+    let other_traits = ["Sized"];
```

Comment:
Lift these into a `const` (but not `pub const` -- not public API)? It would be nice to have top-level items that we can refer to in documentation that make it easy to see which traits we manually include and pretend are in the rustdoc JSON file.

Reply:
Done in 016398d8edbd2f7eb1dc87aebb0561ebb9ec957f.